app/models/search_query.rb

```diff
@@ -79,10 +89,33 @@ def update_for_click(position)
   delegate :available_deadlines, to: ExternalDeadlineFilter
 
   def results
-    results = Case::BasePolicy::Scope.new(User.find(user_id), Case::Base.all).for_view_only
-    results = results.search(search_text)
+    if parent && parent.list_params.present?
+      results = list_results
+    else
+      results = search_results
+    end
     FILTER_CLASSES.reduce(results) do |result, filter_class|
       filter_class.new(self, result).call
     end
   end
+
+  private
+
+  def search_results
+    results = Case::BasePolicy::Scope.new(User.find(user_id), Case::Base.all).for_view_only
+    results.search(search_text)
+  end
+
+  def list_results
+    request = OpenStruct.new(path: list_path, params: YAML.load(list_params))
```

Comment:
Instantiating the `GlobalNavManager` here doesn't feel right. It's already been instantiated correctly in the controller as `@global_nav_manager` so why don't we pass the result of `current_page_or_tab.cases.by_deadline`  all the way into here? It'd have to go through a few methods: `CasesController#open_cases` -> `CasesController#search_and_filter` -> `CaseSearchService#call` -> `SearchQuery#results` -> `SearchQuery#list_results` but none of these take parameters and it'd make sense to have a base AREL passed through to operate on.

Reply:
I see this is part of a larger possible strategy to move search logic from `GlobalNavManager` to `SearchQuery`. LGTM.